app/Models/Semester.php

```diff
@@ -201,10 +201,10 @@ public static function current(): Semester
         if (! Cache::get('semester.current.'.$today)) {
             $now = Carbon::now();
             if ($now->month >= self::START_OF_SPRING_SEMESTER && $now->month <= self::END_OF_SPRING_SEMESTER) {
-                $part = 2;
+                $part = "2"; // Because of some strange reaseon, the part is stored as a string.
                 $year = $now->year - 1;
             } else {
-                $part = 1;
+                $part = "1";
```

Comment:
I'd prefer changing the representation instead

Reply:
And please add this in a different PR.

Reply:
I'll create a new PR, I don't really understand why this is not working currently.